cmd/influxd/upgrade/logging_windows.go

```diff
@@ -0,0 +1,41 @@
+package upgrade
+
+import (
+	"net/url"
+	"os"
+	"path/filepath"
+
+	"go.uber.org/zap"
+)
+
+func init() {
+	// NOTE: Work around a bug in zap's handling of absolute paths on Windows.
+	// See https://github.com/uber-go/zap/issues/621
+	newWinFileSink := func(u *url.URL) (zap.Sink, error) {
+		// Remove leading slash left by url.Parse()
+		return os.OpenFile(u.Path[1:], os.O_WRONLY|os.O_APPEND|os.O_CREATE, 0644)
+	}
+	zap.RegisterSink("winfile", newWinFileSink)
+}
+
+func buildLogger(options *logOptions, verbose bool) (*zap.Logger, error) {
+	config := zap.NewProductionConfig()
+
+	config.Level = zap.NewAtomicLevelAt(options.logLevel)
+	if verbose {
+		config.Level.SetLevel(zap.DebugLevel)
+	}
+
+	logPath := "winfile:///" + filepath.Abs(options.logPath)
```

Comment:
I feel like buildLogger could be unified into one procedure, with a little dependency injection to fix up logPath, like a `generateOsDependentLogPath()` or something (but better-named).

Reply:
Nice idea, will do 👍